.github/workflows/publish.yml

```diff
@@ -4,6 +4,7 @@ on:
   release:
     types: [published]
 
+permissions: {}
```

Comment:
If I understand correctly, this should provide the `contents: read` permissions as well since we use the `actions/checkout@v3` action.